dbt/include/athena/macros/materializations/models/table/table.sql

```diff
@@ -82,27 +82,36 @@
         {{ create_table_as(False, target_relation, sql) }}
       {%- endcall %}
     {%- else -%}
-      {%- if tmp_relation is not none -%}
-        {%- do drop_relation(tmp_relation) -%}
-      {%- endif -%}
-
-      {%- set old_relation_bkp = make_temp_relation(old_relation, '__bkp') -%}
-      -- If we have this, it means that at least the first renaming occurred but there was an issue
-      -- afterwards, therefore we are in weird state. The easiest and cleanest should be to remove
-      -- the backup relation. It won't have an impact because since we are in the else condition,
-      -- that means that old relation exists therefore no downtime yet.
-      {%- if old_relation_bkp is not none -%}
-        {%- do drop_relation(old_relation_bkp) -%}
+      {%- if old_relation.is_view -%}
+        {%- call statement('main') -%}
+          {{ create_table_as(False, tmp_relation, sql) }}
```

Comment:
why not using `{{ create_table_as(False, target_relation, sql) }}` and avoid the final rename?